### PR TITLE
First implementation

### PR DESCRIPTION
**Pull request by pbrisbin**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -1,7 +1,29 @@
...
-description: "Action description"
-author: "Freckle"
+name: Await Statuses
+description: Wait for statuses on a given ref
+author: Freckle
 inputs:
+  ref:
+    description: |
+      The ref to poll status checks on. Can be a branch name (e.g. for PRs) or a
+      SHA (e.g. for pushes). Default is github.ref.
```
Comment (pbrisbin) [inline action.yml:8]: Need to update these docs now. The default is the PR head branch for PRs and `github.ref` otherwise.

File: src/status.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  return {
+    pending,
+    failed,
+    succeeded,
+  };
+}
+
+function includes<T>(xs: Array<T>, x: T): boolean {
+  return xs.indexOf(x) >= 0;
```
Comment (pbrisbin) [inline src/status.ts:75]: Is `!== -1` more idiomatic?

Comment (stackptr) [reply src/status.ts:75]: Yeah

**Review by stackptr — APPROVED** [review]

File: src/status.test.ts [hunk 3]
```diff
@@ -0,0 +1,1940 @@
...
+    "unknown-status",
+  ]);
+
+  expect(statuses.pending).toEqual(["unknown-status"]);
+  expect(statuses.succeeded).toEqual(["backend / test", "test-sql"]);
+  expect(statuses.failed).toEqual(["platform-frontend / platform-image"]);
+});
+
+// Just ran this for some random SHA in a busy project and captured it
+const TEST_CHECK_RUNS = [
```
[the remark this answers is not in the document]
Comment (pbrisbin) [reply src/status.test.ts:19]: I wanted to but didn't know how. Do you think you could commit to this PR to help?

[the remark this answers is not in the document]
Comment (pbrisbin) [reply src/status.test.ts:19]: (I would do `test/files/` or just `files/` I think, `src/` feels wrong.)

[the remark this answers is not in the document]
Comment (pbrisbin) [reply src/status.test.ts:19]: OK, I got this working (tests pass), but there are still two questions:

1. It seems I must put it in `src`. Even if I added to `include`, it doesn't work unless it's in `src`
2. I get an error in my editor (vim + ALE) saying to add `--resolveJsonModule` even though I added it to `tsconfig.json` already (which is required for the tests to pass)

:shrug:

**Review by danroyo — APPROVED** [review]

**Comment by pbrisbin** [comment]
Running one more test from megarepo, then I'll merge.

**Pull request merged**